src/ncp/ncp_base.cpp

```diff
@@ -1210,22 +1210,47 @@ otError NcpBase::CommandHandler_RESET(uint8_t aHeader)
 {
     OT_UNUSED_VARIABLE(aHeader);
 
-    otError error = OT_ERROR_NONE;
+    otError error      = OT_ERROR_NONE;
+    uint8_t reset_type = SPINEL_RESET_STACK;
 
-    // Signal a platform reset. If implemented, this function
-    // shouldn't return.
-    otInstanceReset(mInstance);
+    if (mDecoder.GetRemainingLengthInStruct() > 0)
+    {
+        error = mDecoder.ReadUint8(reset_type);
+        OT_ASSERT(error == OT_ERROR_NONE);
+    }
 
-#if OPENTHREAD_MTD || OPENTHREAD_FTD
-    // We only get to this point if the
-    // platform doesn't support resetting.
-    // In such a case we fake it.
+#if OPENTHREAD_RADIO
+    if (reset_type == SPINEL_RESET_STACK)
+    {
+        mInstance->RadioStackReset();
```

Comment:
I suggest adding a public API for this (probably in `instance.h`), so that we don't need to call method on `Instance` from `NcpBase`

Reply:
Done.